File: languages/cpp/01-foundations/control-flow/example/menu-loop.cpp
#include <iostream>

using namespace std;

int main() {
    int choice = 0;
    int sum = 0;
    int number = 0;

    while (true) {
        cout << "\nMenu\n";
        cout << "1) Add number to sum\n";
        cout << "2) Show current sum\n";
        cout << "3) Reset sum\n";
        cout << "4) Exit\n";
        cout << "Choose an option: ";
        cin >> choice;

        switch (choice) {
            case 1:
                cout << "Enter a number: ";
                cin >> number;
                sum += number;
                cout << "Added. New sum is " << sum << ".\n";
                break;
            case 2:
                cout << "Current sum: " << sum << '\n';
                break;
            case 3:
                sum = 0;
                cout << "Sum reset to 0.\n";
                break;
            case 4:
                cout << "Goodbye.\n";
                return 0;
            default:
                cout << "Invalid option. Try again.\n";
                break;
        }
    }
}

File: languages/go/01-foundations/control-flow/example/menu-loop.go
// This extra example extends control-flow with a menu-driven loop.
// Example purpose: show repeated branching and state updates.

package main

import "fmt"

func main() {
	currentSum := 0

	for {
		fmt.Println("\nMenu")
		fmt.Println("1) Add number to sum")
		fmt.Println("2) Show current sum")
		fmt.Println("3) Reset sum")
		fmt.Println("4) Exit")
		fmt.Print("Choose an option: ")

		var choice int
		if _, err := fmt.Scan(&choice); err != nil {
			fmt.Println("Invalid option. Try again.")
			return
		}

		switch choice {
		case 1:
			var number int
			fmt.Print("Enter a number: ")
			if _, err := fmt.Scan(&number); err != nil {
				fmt.Println("Invalid number. Try again.")
				return
			}
			currentSum += number
			fmt.Printf("Added. New sum is %d.\n", currentSum)
		case 2:
			fmt.Printf("Current sum: %d\n", currentSum)
		case 3:
			currentSum = 0
			fmt.Println("Sum reset to 0.")
		case 4:
			fmt.Println("Goodbye.")
			return
		default:
			fmt.Println("Invalid option. Try again.")
		}
	}
}
